torchmultimodal/utils/attention.py

```diff
@@ -25,11 +25,13 @@ def get_extended_attention_mask(attention_mask: Tensor) -> Tensor:
     # We can provide a self-attention mask of dimensions [batch_size, from_seq_length, to_seq_length]
     # ourselves in which case we just need to make it broadcastable to all heads,
     # [batch_size, num_heads, from_seq_length, to_seq_length].
-    if attention_mask.dim() == 3:
+    if attention_mask.dim() == 4:
```

Comment:
I'm not sure if we need this. I suppose people who use this function are aware whether the original masks have the right dims or not.

Reply:
This is added because in the decoder case, `get_extended_attention_mask_for_decoder` is called at a higher stack level, so `get_extended_attention_mask` will essentially be a noop.